contracts/AssetPool.sol

```diff
@@ -274,6 +274,8 @@ contract AssetPool is Ownable {
     }
 
     function _deposit(address depositor, uint256 amount) internal {
+        require(depositor != address(this), "Self-deposit not allowed");
```

Comment:
Do we want to add a test for this check?

Reply:
We could and I was thinking about it as well but... calling this function so that `depositor == address(this)` is currently not possible, and it is tested [here](https://github.com/keep-network/coverage-pools/blob/c205f56836b8208c081becd31c7d08c920273ae5/test/AssetPool.test.js#L229-L242). This `require` is actually not needed in this setup but it was an audit recommendation to add it and I agree it makes sense to be very explicit about this requirement. Making this function public in a stub to test this requirement... I am not sure if it's worth it. Public interface is protected anyway.